spec/src/main/asciidoc/ch04-query-language.adoc

```diff
@@ -705,51 +734,93 @@ join_collection_valued_path_expression ::=
 join_single_valued_path_expression ::=
     identification_variable.{single_valued_embeddable_object_field.}*single_valued_object_field
 
-join_spec ::= [LEFT [OUTER] | INNER] JOIN
-
 join_condition ::= ON conditional_expression
 ----
 
-The inner and outer join operation types
-described in <<a4884>>, <<a4898>>, and <<a4931>> are supported.
+The inner and outer join operation types described in <<a4884>>, <<a4898>>,
+and <<a4931>> are supported.
+
+===== Inner Joins [[a4884]]
+
+The syntax for an inner join to an entity type is given by:
+
+----
+[INNER] JOIN range_variable_declaration [join_condition]
+----
+
+The keyword INNER is optional and does not affect the semantics
+of the query.
+
+[source,sql]
+----
+SELECT c
+FROM Customer c
+    JOIN Order o ON o.customer.id = c.id
+WHERE c.status = 1
+----
+
+Or, equivalently:
+
+[source,sql]
+----
+SELECT c
+FROM Customer c
+    INNER JOIN Order o ON o.customer.id = c.id
+WHERE c.status = 1
+----
+
+These queries are equivalent to the following query involving
+an implicit "theta" join:
 
-===== Inner Joins (Relationship Joins) [[a4884]]
+[source,sql]
+----
+SELECT c
+FROM Customer c, Order o
+WHERE o.customer.id = c.id AND c.status = 1
+----
 
-The syntax for the inner join operation is
+The syntax for an inner join over an association is given by:
 
 ----
 [INNER] JOIN join_association_path_expression [AS] identification_variable [join_condition]
 ----
 
-For example, the query below joins over the
-relationship between customers and orders. This type of join typically
-equates to a join over a foreign key relationship in the database.
+For example, the query below joins over the relationship between
+customers and orders. This type of join typically equates to a
+join over a foreign key relationship in the database.
 
 [source,sql]
 ----
-SELECT c FROM Customer c JOIN c.orders o WHERE c.status = 1
+SELECT c
+FROM Customer c
+   JOIN c.orders o
+WHERE c.status = 1
 ----
 
-The keyword INNER may optionally be used:
+Equivalently:
 
 [source,sql]
 ----
-SELECT c FROM Customer c INNER JOIN c.orders o WHERE c.status = 1
+SELECT c
+FROM Customer c
+    INNER JOIN c.orders o
+WHERE c.status = 1
 ----
 
-This is equivalent to the following query
-using the earlier IN construct, defined in
-<<a19497>>. It selects those customers of
+This is equivalent to the following query using the earlier IN
+construct, defined in <<a19497>>. It selects those customers of
 status 1 for which at least one order exists:
 
 [source,sql]
 ----
-SELECT OBJECT(c) FROM Customer c, IN(c.orders) o WHERE c.status = 1
+SELECT OBJECT(c)
+FROM Customer c, IN(c.orders) o
```

Comment:
Can we please deprecate this syntax? It violates the isolation of "from clause spaces". SQL says that these "spaces", separated by a comma, are "isolated" i.e. you can't refer to aliases of a different space. The `collection_member_declaration` rule violates this constraint though and for IMO no good reason. People can just as well write an inner join instead.

Reply:
I have no absolutely no objection to explicitly deprecating it. It was _effectively_ already deprecated by JPA 1.0 (it's a holdover from the older EJB-QL).

But let's make that a separate issue.